packages/protocol-kit/src/utils/passkeys/PasskeySigner.ts

```diff
@@ -0,0 +1,238 @@
+
+// Sepolia only
+const P256_VERIFIER_ADDRESS = '0xcA89CBa4813D5B40AeC6E57A30d0Eeb500d6531b' // FCLP256Verifier
+
+/**
+ * Represents a Signer that is created using a passkey.
+ * This class extends the AbstractSigner to implement signer functionalities.
+ *
+ * @extends {AbstractSigner}
+ */
+class PasskeySigner extends AbstractSigner {
+  /**
+   * The raw identifier of the passkey.
+   * see: https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential/rawId
+   */
+  passkeyRawId: ArrayBuffer
+
+  /**
+   * Passkey Coordinates.
+   */
+  coordinates: PasskeyCoordinates
+
+  /**
+   * Safe WebAuthn signer factory Contract.
+   */
+  safeWebAuthnSignerFactoryContract: SafeWebAuthnSignerFactoryContractImplementationType
+
+  constructor(
+    passkeyRawId: ArrayBuffer,
+    coordinates: PasskeyCoordinates,
+    safeWebAuthnSignerFactoryContract: SafeWebAuthnSignerFactoryContractImplementationType,
+    provider: Provider
+  ) {
+    super(provider)
+
+    this.passkeyRawId = passkeyRawId
+    this.coordinates = coordinates
+    this.safeWebAuthnSignerFactoryContract = safeWebAuthnSignerFactoryContract
+  }
+
+  static async init(
+    passkey: passkeyArgType,
+    safeWebAuthnSignerFactoryContract: SafeWebAuthnSignerFactoryContractImplementationType,
+    provider: Provider
+  ): Promise<PasskeySigner> {
+    const coordinates = await extractPasskeyCoordinates(passkey.publicKey)
+
+    return new PasskeySigner(
+      passkey.rawId,
+      coordinates,
+      safeWebAuthnSignerFactoryContract,
+      provider
+    )
+  }
+
+  /**
+   * Returns the address associated with the Signer.
+   * @returns {Promise<string>} A promise that resolves to the signer's address.
+   */
+  async getAddress(): Promise<string> {
+    const [signerAddress] = await this.safeWebAuthnSignerFactoryContract.getSigner([
+      BigInt(this.coordinates.x),
+      BigInt(this.coordinates.y),
+      BigInt(P256_VERIFIER_ADDRESS)
+    ])
+
+    return signerAddress
+  }
+
+  /**
+   * Encodes the createSigner contract function.
+   * @returns {string} The encoded data to create a signer.
+   */
+  encodeCreateSigner(): string {
+    return this.safeWebAuthnSignerFactoryContract.encode('createSigner', [
+      BigInt(this.coordinates.x),
+      BigInt(this.coordinates.y),
+      BigInt(P256_VERIFIER_ADDRESS)
+    ])
+  }
+
+  /**
+   * Signs the provided data using the passkey.
+   * @param {Uint8Array} data - The data to be signed.
+   * @returns {Promise<string>} A promise that resolves to the signed data.
+   */
+  async sign(data: Uint8Array): Promise<string> {
+    const assertion = (await navigator.credentials.get({
+      publicKey: {
+        challenge: data,
+        allowCredentials: [{ type: 'public-key', id: this.passkeyRawId }],
+        userVerification: 'required'
+      }
+    })) as PublicKeyCredential & { response: AuthenticatorAssertionResponse }
+
+    if (!assertion || !assertion?.response?.authenticatorData) {
```

Comment:
I dont think you need the first verification since you are using safe navigation/elvis operator.

Reply:
done!